crates/re_query/src/range/results.rs

```diff
@@ -566,6 +566,24 @@ impl RangeComponentResults {
             }
         };
 
+        // # Reentrancy edge-case
+        //
+        // If we are in the reentrancy case, and if it's the first time this cache is used at all, and if
+        // the previous stack-frame that was holding the lock finally decided to release it without
+        // actually caching anything, then the deserialization cache still won't be initialized.
+        //
+        // Just leave it be for now, it'll fix itself by next frame.
+        if results.cached_dense.is_none() {
+            return RangeData {
+                indices: None,
+                data: None,
+                time_range: TimeRange::EMPTY,
+                front_status: PromiseResult::Ready(()),
+                back_status: PromiseResult::Ready(()),
+                reentering: &REENTERING,
+            };
+        }
```

Comment:
Is there a possibility for some other thread to do something bad between this `if` and the the unwrap below?

Reply:
Not that I know of; but I'd like to automate as much as possible the reentrancy checklist entry in a follow up PR, now that we can programmatically set time ranges.